### PR TITLE
fix: typos in documentation files

### PR DESCRIPTION
**Pull request by leopardracer**
This pull request contains changes to improve clarity, correctness and structure.

**Description correction:**
Corrected "gitub-mgmt" to "GitHub-mgmt".
Corrected "expecations" to "expectations".
Corrected "insallations" to "installations".
Corrected "Confrim" to "Confirm".
Corrected "modifes" to "modifies".
Corrected "worfklows" to "workflows".

Please review the changes and let me know if any additional changes are needed.

### CODE REVIEW

**Comment by leopardracer** [comment]
@Stebalien Hi, Is everything okay or do I need to fix something?

**Comment by Stebalien** [comment]
Sorry, I thought this was a PR against a different repo.

File: README.md [hunk 1]
```diff
@@ -15,22 +15,22 @@ To learn more, check out:
 ## Tips / FYIs
 * [github/filecoin-project.yml](https://github.com/filecoin-project/github-mgmt/blob/master/github/filecoin-project.yml) is the key file where updates are made to adjust permissions.
 * "github-mgmt" was the old name.  The original template repo is now called "github-as-code".  We use the terms interchangably in comments/docs. 
-* Yes, it's confusing to have a `.github` and `github` directory.  That is how gitub-mgmt was originally setup an we're living with it.  (At least of 2024-09-05, there is [no backlog item for cleaning this up](https://github.com/ipdxco/github-as-code/issues?page=1&q=is%3Aissue+is%3Aopen).)
+* Yes, it's confusing to have a `.github` and `github` directory.  That is how GitHub-mgmt was originally setup an we're living with it.  (At least of 2024-09-05, there is [no backlog item for cleaning this up](https://github.com/ipdxco/github-as-code/issues?page=1&q=is%3Aissue+is%3Aopen).)
```
Comment (Stebalien) [inline README.md:18]:
```suggestion
* Yes, it's confusing to have a `.github` and `github` directory.  That is how `github-mgmt` was originally setup an we're living with it.  (At least of 2024-09-05, there is [no backlog item for cleaning this up](https://github.com/ipdxco/github-as-code/issues?page=1&q=is%3Aissue+is%3Aopen).)
```

Comment (leopardracer) [reply README.md]: Corrected!

**Review by Stebalien — APPROVED** [review]

**Review by BigLep — APPROVED** [review]
I'll approve / merge.  Some of these changes should be made in https://github.com/ipdxco/github-as-code where this was forked/synced from.

**Pull request merged**